packages/neon/neon/lib/src/utils/request_manager.dart

```diff
@@ -34,26 +38,31 @@ class RequestManager {
 
   Cache? _cache;
 
-  Future<void> wrapNextcloud<T, R>(
+  Future<void> wrapNextcloud<T, R, H>(
     final String clientID,
     final String k,
     final BehaviorSubject<Result<T>> subject,
-    final NextcloudApiCallback<R> call,
-    final UnwrapCallback<T, R> unwrap, {
+    final DynamiteRawResponse<R, H> rawResponse,
+    final UnwrapCallback<T, DynamiteResponse<R, H>> unwrap, {
     final bool disableTimeout = false,
-    final bool emitEmptyCache = false,
   }) async =>
-      _wrap<T, R>(
+      _wrap<T, DynamiteRawResponse<R, H>>(
         clientID,
         k,
         subject,
-        call,
-        unwrap,
-        (final data) => json.encode(serializeNextcloud<R>(data)),
-        (final data) => deserializeNextcloud<R>(json.decode(data) as Object),
+        () async {
+          await rawResponse.future;
+          return rawResponse;
+        },
+        (final rawResponse) => unwrap(rawResponse.response),
```

Comment:
Just do `unwrap(await rawResponse.future)` again?

Reply:
The problem is that the future contains the deserialized response (DynamiteResponse) but the `_wrap` methods expects a `DynamiteRawResponse` so it can serialize it again for the cache.